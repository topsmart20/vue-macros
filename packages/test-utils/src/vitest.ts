import path from 'node:path'
import glob, { type Options as GlobOptions } from 'fast-glob'
import { describe, expect, test } from 'vitest'
import { normalizePath } from '@rollup/pluginutils'

interface Options {
  params?: [name: string, values?: any[]][]
  promise?: boolean
}

const SKIP_VUE2 = !!process.env.SKIP_VUE2

export async function testFixtures(
  globs: string | string[],
  exec: (args: Record<string, any>, id: string) => any,
  options?: GlobOptions & Options,
): Promise<void>
export async function testFixtures(
  files: Record<string, string>,
  exec: (args: Record<string, any>, id: string, code: string) => any,
  options?: Options,
): Promise<void>
export async function testFixtures(
  globsOrFiles: string | string[] | Record<string, string>,
  cb: (args: Record<string, any>, id: string, code: string) => any,
  { params, promise, ...globOptions }: GlobOptions & Options = {},
) {
  let files: Record<string, string | undefined>

  if (typeof globsOrFiles === 'string' || Array.isArray(globsOrFiles)) {
    files = Object.fromEntries(
      (await glob(globsOrFiles, globOptions)).map((file) => [file, undefined]),
    )
  } else {
    files = globsOrFiles
  }

  for (const [id, code] of Object.entries(files)) {
    makeTests(id, code, [[normalizePath(id)], ...(params || [])])()
  }

  function getName(name: string, value: any) {
    return value !== undefined ? `${name} = ${String(value)}` : name
  }

  function makeTests(
    id: string,
    code: string | undefined,
    params: NonNullable<Options['params']>,
    args: Record<string, any> = {},
  ) {
    const [currParams, ...restParams] = params
    const [name, values = [undefined]] = currParams
    if (restParams.length > 0) {
      return () => {
        for (const value of values) {
          const currArgs = { ...args, [name]: value }

          describe(
            getName(name, value),
            makeTests(id, code, restParams, currArgs),
          )
        }
      }
    } else {
      return () => {
        for (const value of values) {
          const testName = getName(name, value)
          const isSkip = testName.includes('vue2') && SKIP_VUE2

          test.skipIf(isSkip)(testName, async () => {
            const currArgs = { ...args, [name]: value }
            const execute = () =>
              cb(currArgs, path.resolve(globOptions.cwd || '/', id), code!)
            if (id.includes('error')) {
              if (promise) {
                await expect(execute()).rejects.toThrowErrorMatchingSnapshot()
              } else {
                expect(execute).toThrowErrorMatchingSnapshot()
              }
            } else if (promise) {
              await expect(
                (execute() as Promise<any>).catch((error) => {
                  console.warn(error)
                  return Promise.reject(error)
                }),
              ).resolves.toMatchSnapshot()
            } else {
              expect(execute()).toMatchSnapshot()
            }
          })
        }
      }
    }
  }
}
