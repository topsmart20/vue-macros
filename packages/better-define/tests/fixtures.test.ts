import { resolve } from 'node:path'
import { describe, expect, test } from 'vitest'
import glob from 'fast-glob'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  RollupVueJsx,
  rollupBuild,
} from '@vue-macros/test-utils'
import VueBetterDefine from '../src/rollup'

describe('fixtures', async () => {
  const root = resolve(__dirname, '..')
  const files = await glob(
    ['tests/fixtures/*.vue', '!tests/fixtures/*.exclude.vue'],
    {
      cwd: root,
      onlyFiles: true,
    }
  )

  for (const file of files) {
    describe(file.replace(/\\/g, '/'), () => {
      const filepath = resolve(root, file)

      for (const isProduction of [true, false]) {
        test(`isProduction is ${isProduction}`, async () => {
          const code = await rollupBuild(filepath, [
            VueBetterDefine({ isProduction }),
            RollupVue(),
            RollupVueJsx(),
            RollupRemoveVueFilePathPlugin(),
            RollupEsbuildPlugin({
              target: 'esnext',
            }),
          ])
          expect(code).toMatchSnapshot()
        })
      }
    })
  }
})
