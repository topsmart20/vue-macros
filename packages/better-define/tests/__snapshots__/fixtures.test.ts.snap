// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue > isProduction = false 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"basic\\",
  props: {
    \\"base\\": { type: String, required: true },
    \\"str\\": { type: String, required: true },
    \\"num\\": { type: Number, required: true },
    \\"map\\": { type: Map, required: true },
    \\"arr\\": { type: Array, required: true },
    \\"union\\": { type: [String, Number], required: true }
  },
  emits: [\\"click\\", \\"change\\"],
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/basic.vue > isProduction = true 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"basic\\",
  props: {
    \\"base\\": null,
    \\"str\\": null,
    \\"num\\": null,
    \\"map\\": null,
    \\"arr\\": null,
    \\"union\\": null
  },
  emits: [\\"click\\", \\"change\\"],
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/defaults-dynamic.vue > isProduction = false 1`] = `
"import { defineComponent, mergeDefaults, openBlock, createElementBlock } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"defaults-dynamic\\",
  props: mergeDefaults({
    \\"foo\\": { type: String, required: false }
  }, {
    [\\"foo\\"]: \\"foo\\"
  }),
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(\\"div\\");
    };
  }
});

var defaultsDynamic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { defaultsDynamic as default };
"
`;

exports[`fixtures > tests/fixtures/defaults-dynamic.vue > isProduction = true 1`] = `
"import { defineComponent, mergeDefaults, openBlock, createElementBlock } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"defaults-dynamic\\",
  props: mergeDefaults({
    \\"foo\\": null
  }, {
    [\\"foo\\"]: \\"foo\\"
  }),
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(\\"div\\");
    };
  }
});

var defaultsDynamic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { defaultsDynamic as default };
"
`;

exports[`fixtures > tests/fixtures/defaults-static.vue > isProduction = false 1`] = `
"import { defineComponent, openBlock, createElementBlock } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"defaults-static\\",
  props: {
    \\"foo\\": { type: String, required: false, default: \\"foo\\" },
    \\"bar\\": { type: Number, required: false, get default() {
      return 10;
    } },
    \\"baz\\": { type: Promise, required: false, async default() {
      return 10;
    } },
    \\"qux\\": { type: Function, required: false, default: () => {
    } },
    \\"quux\\": { required: false, default: abc }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(\\"div\\");
    };
  }
});

var defaultsStatic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { defaultsStatic as default };
"
`;

exports[`fixtures > tests/fixtures/defaults-static.vue > isProduction = true 1`] = `
"import { defineComponent, openBlock, createElementBlock } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"defaults-static\\",
  props: {
    \\"foo\\": { default: \\"foo\\" },
    \\"bar\\": { get default() {
      return 10;
    } },
    \\"baz\\": { async default() {
      return 10;
    } },
    \\"qux\\": { type: Function, default: () => {
    } },
    \\"quux\\": { default: abc }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(\\"div\\");
    };
  }
});

var defaultsStatic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { defaultsStatic as default };
"
`;

exports[`fixtures > tests/fixtures/fn-default.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"fn-default\\",
  props: {
    \\"fn\\": { type: Function, required: true, default: () => {
    } }
  },
  setup(__props) {
    const props = __props;
    return () => {
    };
  }
});

var fnDefault = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { fnDefault as default };
"
`;

exports[`fixtures > tests/fixtures/fn-default.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"fn-default\\",
  props: {
    \\"fn\\": { type: Function, default: () => {
    } }
  },
  setup(__props) {
    const props = __props;
    return () => {
    };
  }
});

var fnDefault = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { fnDefault as default };
"
`;

exports[`fixtures > tests/fixtures/import-from-vue.vue > isProduction = false 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"import-from-vue\\",
  props: {
    \\"setupBase\\": { type: String, required: true },
    \\"num\\": { type: Number, required: true },
    \\"bool\\": { type: Boolean, required: true },
    \\"map\\": { type: Map, required: true },
    \\"arr\\": { type: Array, required: true },
    \\"union\\": { type: [String, Number], required: true },
    \\"normalBase\\": { type: String, required: true },
    \\"str\\": { type: String, required: true },
    \\"setupBase_2\\": { type: String, required: true }
  },
  emits: [\\"click\\", \\"change\\"],
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var importFromVue = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { importFromVue as default };
"
`;

exports[`fixtures > tests/fixtures/import-from-vue.vue > isProduction = true 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"import-from-vue\\",
  props: {
    \\"setupBase\\": null,
    \\"num\\": null,
    \\"bool\\": { type: Boolean },
    \\"map\\": null,
    \\"arr\\": null,
    \\"union\\": null,
    \\"normalBase\\": null,
    \\"str\\": null,
    \\"setupBase_2\\": null
  },
  emits: [\\"click\\", \\"change\\"],
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var importFromVue = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { importFromVue as default };
"
`;

exports[`fixtures > tests/fixtures/intersection.vue > isProduction = false 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"intersection\\",
  props: {
    \\"base\\": { type: String, required: true },
    \\"str\\": { type: String, required: true },
    \\"num\\": { type: Number, required: true },
    \\"map\\": { type: Map, required: true },
    \\"arr\\": { type: Array, required: true },
    \\"union\\": { type: [String, Number], required: true }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var intersection = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { intersection as default };
"
`;

exports[`fixtures > tests/fixtures/intersection.vue > isProduction = true 1`] = `
"import { defineComponent, renderSlot } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"intersection\\",
  props: {
    \\"base\\": null,
    \\"str\\": null,
    \\"num\\": null,
    \\"map\\": null,
    \\"arr\\": null,
    \\"union\\": null
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, \\"default\\");
    };
  }
});

var intersection = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { intersection as default };
"
`;

exports[`fixtures > tests/fixtures/issue-260.vue > isProduction = false 1`] = `
"import { defineComponent, toDisplayString } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-260\\",
  props: {
    \\"menu\\": { type: [Object, Array], required: false, default: () => ({ txt: \\"1-1\\" }) }
  },
  setup(__props) {
    const props = __props;
    return (_ctx, _cache) => {
      return toDisplayString(props.menu);
    };
  }
});

var issue260 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue260 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-260.vue > isProduction = true 1`] = `
"import { defineComponent, toDisplayString } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-260\\",
  props: {
    \\"menu\\": { default: () => ({ txt: \\"1-1\\" }) }
  },
  setup(__props) {
    const props = __props;
    return (_ctx, _cache) => {
      return toDisplayString(props.menu);
    };
  }
});

var issue260 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue260 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-328.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-328\\",
  emits: [\\"foo\\"],
  setup(__props) {
    return () => {
    };
  }
});

var issue328 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue328 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-328.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-328\\",
  emits: [\\"foo\\"],
  setup(__props) {
    return () => {
    };
  }
});

var issue328 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue328 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-351.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-351\\",
  props: {
    \\"modelValue\\": { type: [String, Boolean], required: true }
  },
  setup(__props) {
    return () => {
    };
  }
});

var issue351 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue351 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-351.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-351\\",
  props: {
    \\"modelValue\\": { type: [String, Boolean] }
  },
  setup(__props) {
    return () => {
    };
  }
});

var issue351 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue351 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-362.vue > isProduction = false 1`] = `
"import { defineComponent, openBlock, createElementBlock, toDisplayString } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-362\\",
  props: {
    \\"trigger\\": { required: true }
  },
  setup(__props) {
    const props = __props;
    const Trigger = {
      click: \\"click\\",
      focus: \\"focus\\",
      hover: \\"hover\\"
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        \\"button\\",
        null,
        toDisplayString(__props.trigger),
        1
        /* TEXT */
      );
    };
  }
});

var issue362 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue362 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-362.vue > isProduction = true 1`] = `
"import { defineComponent, openBlock, createElementBlock, toDisplayString } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"issue-362\\",
  props: {
    \\"trigger\\": null
  },
  setup(__props) {
    const props = __props;
    const Trigger = {
      click: \\"click\\",
      focus: \\"focus\\",
      hover: \\"hover\\"
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        \\"button\\",
        null,
        toDisplayString(__props.trigger),
        1
        /* TEXT */
      );
    };
  }
});

var issue362 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue362 as default };
"
`;

exports[`fixtures > tests/fixtures/optional-method.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"optional-method\\",
  props: {
    \\"onClick\\": { type: Function, required: false },
    \\"onSubmit\\": { type: Function, required: false }
  },
  setup(__props) {
    return () => {
    };
  }
});

var optionalMethod = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { optionalMethod as default };
"
`;

exports[`fixtures > tests/fixtures/optional-method.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"optional-method\\",
  props: {
    \\"onClick\\": { type: Function },
    \\"onSubmit\\": { type: Function }
  },
  setup(__props) {
    return () => {
    };
  }
});

var optionalMethod = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { optionalMethod as default };
"
`;

exports[`fixtures > tests/fixtures/resolve-failed.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"resolve-failed\\",
  props: {
    \\"foo\\": { required: true },
    \\"bar\\": { type: Object, required: true },
    \\"bool\\": { type: Boolean, skipCheck: true, required: true },
    \\"fun\\": { type: Function, skipCheck: true, required: true },
    \\"boolAndFun\\": { type: [Boolean, Function], skipCheck: true, required: true }
  },
  setup(__props) {
    return () => {
    };
  }
});

var resolveFailed = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { resolveFailed as default };
"
`;

exports[`fixtures > tests/fixtures/resolve-failed.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"resolve-failed\\",
  props: {
    \\"foo\\": null,
    \\"bar\\": null,
    \\"bool\\": { type: Boolean },
    \\"fun\\": { type: Function },
    \\"boolAndFun\\": { type: [Boolean, Function] }
  },
  setup(__props) {
    return () => {
    };
  }
});

var resolveFailed = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { resolveFailed as default };
"
`;

exports[`fixtures > tests/fixtures/special-key.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"special-key\\",
  props: {
    \\"key-name\\": { type: String, required: false }
  },
  setup(__props) {
    return () => {
    };
  }
});

var specialKey = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { specialKey as default };
"
`;

exports[`fixtures > tests/fixtures/special-key.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"special-key\\",
  props: {
    \\"key-name\\": null
  },
  setup(__props) {
    return () => {
    };
  }
});

var specialKey = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { specialKey as default };
"
`;

exports[`fixtures > tests/fixtures/ts-utility-types.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"ts-utility-types\\",
  props: {
    \\"foo\\": { type: String, required: true },
    \\"bar\\": { type: String, required: true }
  },
  setup(__props) {
    return () => {
    };
  }
});

var tsUtilityTypes = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { tsUtilityTypes as default };
"
`;

exports[`fixtures > tests/fixtures/ts-utility-types.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"ts-utility-types\\",
  props: {
    \\"foo\\": null,
    \\"bar\\": null
  },
  setup(__props) {
    return () => {
    };
  }
});

var tsUtilityTypes = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { tsUtilityTypes as default };
"
`;

exports[`fixtures > tests/fixtures/union.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union\\",
  props: {
    \\"type\\": { type: String, required: true },
    \\"foo\\": { type: String, required: false },
    \\"optional\\": { type: Boolean, required: false },
    \\"bar\\": { type: Number, required: false }
  },
  setup(__props) {
    return () => {
    };
  }
});

var union = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { union as default };
"
`;

exports[`fixtures > tests/fixtures/union.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union\\",
  props: {
    \\"type\\": null,
    \\"foo\\": null,
    \\"optional\\": { type: Boolean },
    \\"bar\\": null
  },
  setup(__props) {
    return () => {
    };
  }
});

var union = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { union as default };
"
`;

exports[`fixtures > tests/fixtures/union-emits.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union-emits\\",
  emits: [\\"some\\", \\"emit\\", \\"another\\"],
  setup(__props, { emit }) {
    return () => {
    };
  }
});

var unionEmits = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unionEmits as default };
"
`;

exports[`fixtures > tests/fixtures/union-emits.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union-emits\\",
  emits: [\\"some\\", \\"emit\\", \\"another\\"],
  setup(__props, { emit }) {
    return () => {
    };
  }
});

var unionEmits = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unionEmits as default };
"
`;

exports[`fixtures > tests/fixtures/union-props.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union-props\\",
  props: {
    \\"foo\\": { type: [String, Number], required: true },
    \\"bar\\": { type: [String, Number], required: true },
    \\"optional\\": { type: Boolean, required: false }
  },
  setup(__props) {
    return () => {
    };
  }
});

var unionProps = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unionProps as default };
"
`;

exports[`fixtures > tests/fixtures/union-props.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"union-props\\",
  props: {
    \\"foo\\": null,
    \\"bar\\": null,
    \\"optional\\": { type: Boolean }
  },
  setup(__props) {
    return () => {
    };
  }
});

var unionProps = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unionProps as default };
"
`;

exports[`fixtures > tests/fixtures/unresolved.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"unresolved\\",
  props: {
    \\"foo\\": { required: true }
  },
  setup(__props) {
    return () => {
    };
  }
});

var unresolved = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unresolved as default };
"
`;

exports[`fixtures > tests/fixtures/unresolved.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"unresolved\\",
  props: {
    \\"foo\\": null
  },
  setup(__props) {
    return () => {
    };
  }
});

var unresolved = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { unresolved as default };
"
`;

exports[`fixtures > tests/fixtures/vue-core-7553.vue > isProduction = false 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"vue-core-7553\\",
  props: {
    \\"size\\": { type: String, required: false },
    \\"color\\": { type: String, required: true },
    \\"appearance\\": { type: String, required: true },
    \\"note\\": { type: String, required: false }
  },
  setup(__props) {
    return () => {
    };
  }
});

var vueCore7553 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { vueCore7553 as default };
"
`;

exports[`fixtures > tests/fixtures/vue-core-7553.vue > isProduction = true 1`] = `
"import { defineComponent } from 'vue';
import _export_sfc from '/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: \\"vue-core-7553\\",
  props: {
    \\"size\\": null,
    \\"color\\": null,
    \\"appearance\\": null,
    \\"note\\": null
  },
  setup(__props) {
    return () => {
    };
  }
});

var vueCore7553 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { vueCore7553 as default };
"
`;
